### PR TITLE
fix: render THORChain LP deposits with Pool/Paired Address

### LINKED ISSUE #3781: [Fix] Windows joiner displays THORChain LP deposit as a swap (missing Pool, Paired Address)
# [Fix] Windows joiner displays THORChain LP deposit as a swap

## Problem
When the Windows app acts as a **joining device** for a THORChain LP deposit initiated from another Vultisig client (e.g., iOS initiating `+:ETH.USDC-0XA0B86991…:thor12a9rpf9…`), the Windows Join Keysign / Verify screen misrenders the transaction as a **swap**:

- Header: `You're swapping`
- Displays `0.3 USDC → 0.3 USDC` (same token both sides — there is no swap destination for an LP add).
- Shows `Provider: THORChain` and `Network fee`.
- Does **not** show the Pool, Paired Address, or raw Memo.

The transaction is actually a THORChain LP add (memo prefix `+:<POOL>:<PAIRED_ADDRESS>`), not a swap.

## Expected Behavior
When scanning an LP deposit keysign payload, Windows should render the same fields the initiator shows on the originating client:

- Amount title reflects the LP operation (e.g., `0.3 USDC → ETH.USDC LP`).
- Pool row.
- Paired Address row (when present in the memo).
- Memo row.
- Network and Est. network fee.
- No `You're swapping` header, no `Provider: THORChain` swap framing.

## Steps to Reproduce
1. On another Vultisig client (e.g., iOS), initiate a THORChain LP add to `ETH.USDC` and proceed to the Verify → pair screen.
2. On Windows, scan the pairing QR and navigate to the Join Keysign verify screen.
3. Observe: Windows shows `You're swapping 0.3 USDC → 0.3 USDC`, `Provider: THORChain` — no Pool / Paired Address / Memo rows.

## Likely Cause
The keysign payload's memo format (`+:POOL[:PAIRED_ADDRESS]`) is being classified as a THORChain swap on the Windows side rather than as an LP deposit. The transaction-type detection / UI branching on the joiner likely treats any THORChain deposit with a memo as a swap, and the swap display then picks the same source coin for both sides because no destination asset is encoded (LP adds have no output asset — just pool units).

## Suggested Fix
- On the joiner, detect the `+:POOL…` (add LP) and `-:POOL:BASIS_POINTS` (remove LP) memo prefixes and route to an LP-specific summary view instead of the swap summary.
- Display: amount title with pool destination, Pool row, Paired Address row (when memo has three segments), Memo row, Network, Est. network fee.
- Mirror the initiator's presentation for parity.

## Related
- iOS sibling issue: https://github.com/vultisig/vultisig-ios/issues/4189 (same class of bug — iOS joiner drops LP fields and LP-aware amount title; parity problem across platforms).

## Acceptance Criteria
- [ ] Windows joiner no longer shows `You're swapping` for THORChain LP add/remove payloads.
- [ ] Pool and (when present) Paired Address rows render.
- [ ] Memo is shown.
- [ ] Amount title reflects the pool destination.
- [ ] Non-LP THORChain swaps continue to render correctly.

### PR DESCRIPTION
**Pull request by Ehsan-saradar**
## Summary
- Joiner verify screen used `swapPayload` presence as the only signal, so iOS-initiated THORChain LP add/remove payloads (which reuse `thorchainSwapPayload`) rendered as `You're swapping X → X` with no Pool, Paired Address, or Memo rows.
- Detect LP memos (`+:POOL[:PAIRED_ADDRESS]` / `-:POOL:BASIS_POINTS`) on the joiner and route them to a deposit-style view via a new `JoinKeysignTransactionVerify` router. Memo detection wins over `swapPayload`, so the fix is correct regardless of how the initiator builds the payload.
- Added `pool` / `paired_address` i18n keys (en only — `yarn translate` propagates).

Closes #3781

## Test plan
- [x] `yarn check` (typecheck + lint:fix + knip) passes
- [x] `parseThorLpMemo` unit tests pass (5/5)
- [ ] Manual: iOS initiates THORChain LP add to ETH.USDC → Windows joiner shows Pool, Paired Address, Memo, Amount, Network fee (no `You're swapping`)
- [ ] Manual: iOS initiates LP remove → Windows joiner shows Pool, Memo, Amount, Network fee
- [ ] Manual: iOS initiates a regular THORChain swap → Windows joiner still renders the swap view
- [ ] Manual: non-THORChain sends and other deposits unaffected

## Summary by CodeRabbit

* **New Features**
  * Improved keysign verification: LP (add/remove) transactions are now detected and shown with a dedicated LP verification screen; other transaction types route to appropriate verification views.

* **Localization**
  * Added English labels for "pool" and "paired address" in the UI.

* **Tests**
  * Added unit tests to validate LP memo parsing and edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/vultisig/vultisig-windows/pull/3891)

### CODE REVIEW

File: core/ui/mpc/keysign/join/tx/JoinKeysignLpVerify.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import { useTranslation } from 'react-i18next'
+
+import { JoinKeysignTxPrimaryInfo } from './JoinKeysignTxPrimaryInfo'
+import { ThorLpMemo } from './parseThorLpMemo'
+
+type Props = ValueProp<KeysignPayload> & {
+  lp: ThorLpMemo
+}
+
+export const JoinKeysignLpVerify = ({ value, lp }: Props) => {
```
[the remark this answers is not in the document]
Comment (Ehsan-saradar) [reply core/ui/mpc/keysign/join/tx/JoinKeysignLpVerify.tsx:22]: Done in f6cca85ae — added JSDoc above the component.

File: core/ui/mpc/keysign/join/tx/JoinKeysignTransactionVerify.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { TxOverviewPanel } from '@core/ui/chain/tx/TxOverviewPanel'
+import { ValueProp } from '@lib/ui/props'
+import { KeysignPayload } from '@vultisig/core-mpc/types/vultisig/keysign/v1/keysign_message_pb'
+
+import { JoinKeysignLpVerify } from './JoinKeysignLpVerify'
+import { JoinKeysignSwapVerify } from './JoinKeysignSwapVerify'
+import { JoinKeysignTxPrimaryInfo } from './JoinKeysignTxPrimaryInfo'
+import { parseThorLpMemo } from './parseThorLpMemo'
+
+export const JoinKeysignTransactionVerify = ({
```
[the remark this answers is not in the document]
Comment (Ehsan-saradar) [reply core/ui/mpc/keysign/join/tx/JoinKeysignTransactionVerify.tsx:16]: Done in f6cca85ae — added JSDoc describing the LP/swap/default routing behavior.

File: core/ui/mpc/keysign/join/tx/parseThorLpMemo.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+export type ThorLpMemo =
+  | { kind: 'add'; pool: string; pairedAddress?: string }
+  | { kind: 'remove'; pool: string; basisPoints: number }
+
+export const parseThorLpMemo = (memo: string): ThorLpMemo | null => {
```
[the remark this answers is not in the document]
Comment (Ehsan-saradar) [reply core/ui/mpc/keysign/join/tx/parseThorLpMemo.ts:16]: Done in f6cca85ae — added JSDoc on both `ThorLpMemo` and `parseThorLpMemo`.

File: core/ui/mpc/keysign/join/tx/parseThorLpMemo.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      kind: 'add',
+      pool,
+      pairedAddress: third || undefined,
+    }
+  }
+
+  if (prefix === '-') {
+    const basisPoints = Number(third)
+    if (!Number.isFinite(basisPoints)) return null
+    return { kind: 'remove', pool, basisPoints }
```
[the remark this answers is not in the document]
Comment (Ehsan-saradar) [reply core/ui/mpc/keysign/join/tx/parseThorLpMemo.ts:38]: Done in f6cca85ae — basisPoints now requires `Number.isInteger` and is bounded to `[0, 10000]`. Added tests for negative, out-of-range, and non-integer cases.

**Review by johnnyluo — APPROVED** [review]
LGTM

**Review by rcoderdev — APPROVED** [review]
